internal/listener/listener.go

```diff
@@ -48,7 +48,14 @@ func (l *Listener) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 func (l *Listener) Run() error {
 	l.logger.Infof("Starting listener on http://%s", l.configFile.Listen)
-	return http.ListenAndServe(l.configFile.Listen, l)
+	server := &http.Server{
+		Addr:         l.configFile.Listen,
+		Handler:      l,
+		ReadTimeout:  3 * time.Second,
+		WriteTimeout: 3 * time.Second,
```

Comment:
I think the goal is to have a timeout at all, not a very tight one. I guess all values are pretty much chosen by a fair dice roll, but I'd go for maybe 10 seconds.

Reply:
I just thought three seconds would be a very long time..